src/Microsoft.AspNetCore.Routing/Matching/Ascii.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.AspNetCore.Routing.Matching
+{
+    internal static class Ascii
+    {
+        // case-sensitive equality comparison when we KNOW that 'a' is in the ASCII range
+        // and we know that the spans are the same length.
+        //
+        // Similar to https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs#L549
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool AsciiIgnoreCaseEquals(ReadOnlySpan<char> a, ReadOnlySpan<char> b, int length)
+        {
+            if (a.Length < length || b.Length < length)
+            {
+                // This should never happen, but we don't want to have undefined
+                // behavior if it does.
+                ThrowArgumentException();
+            }
+
+            // We're doing unsafe operations here, caller is expected to have already checked
+            // length.
+            ref char charA = ref MemoryMarshal.GetReference(a);
```

Comment:
Curious: what was the type when it was var?

Reply:
it's char, changed it.